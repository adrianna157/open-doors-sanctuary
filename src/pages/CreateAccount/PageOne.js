import React, { useContext } from "react"
import { validatePageOne } from '../../helpers/createAccount/createAccountValidation.js'
import HeaderText from '../../components/HeaderText'
import Button from '../../components/Button'
import TextField from '../../components/TextField'
import Page from '../../components/Page'
import Container from '../../components/Container'
import { useNavigate } from "react-router-dom"
import { NotificationContext } from "../../helpers/AlertContext/AlertContext.js";
import AlertHandler from "../../components/AlertHandler/index.js"


const PageOne = (props) => {
  const [alert, setAlert] = useContext(NotificationContext);
  let navigate = useNavigate();
  let {
    firstName, setFirstName,
    lastName, setLastName,
    email, setEmail,
    password, setPassword,
    confirmPassword, setConfirmPassword,
    pageChange
  } = props

  return (
    <Page noLeftMargin>
      <AlertHandler />
      <Container
        grayedBackground
        height="md:h-[32rem] sm:h-1/6 lg:h-63pr"
        width="md:w-[32rem] sm:w-1/6 lg:w-1/3"
        padding="pt-3pr px-4pr sm:pl-2"
        className="flex flex-col"
        margin=" ml-16"
        mariginTop="mt-24 lg:mt-h26pr"
      >
        <center>
          <HeaderText noBold fontSize="text-4xl" className="pb-2pr text-[#a26360]">
            Create an Account
          </HeaderText>
        </center>
        <div>
          <TextField
            value={firstName}
            onChange={(e) => setFirstName(e.target.value)}
            placeHolder="First Name"
            label="First Name"
            className="mb-4"
          />
          <TextField
            value={lastName}
            onChange={(e) => setLastName(e.target.value)}
            placeHolder="Last Name"
            label="Last Name"
            className="mb-4"
          />
          <TextField
            type="email"
            value={email}
            onChange={(e) => setEmail(e.target.value)}
            placeHolder="Email"
            label="Email"
            className="mb-4"
          />

          <TextField
            type="password"
            value={password}
            onChange={(e) => setPassword(e.target.value)}
            placeHolder="Password"
            label="Password"
            className="mb-4"
            info="Your password must be 12 characters long, have at least one number, one upper case, one lower case, and one special character"
          />

          <TextField
            type="password"
            value={confirmPassword}
            onChange={(e) => setConfirmPassword(e.target.value)}
            placeHolder="Password"
            label="Confirm Password"
            className="mb-4pr"
          />
          <div className="flex justify-between mb-h2pr">
            <Button
              solidGreen
              noPadding
              className="w-20pr lg:w-6pr"
              onClick={() => {
                navigate(-1);
              }}
            >
              CANCEL
            </Button>
            <Button
              solidGreen
              noPadding
              className="w-20pr lg:w-6pr"
              onClick={() => {
                if (
                  validatePageOne(
                    setAlert,
                    firstName,
                    lastName,
                    email,
                    password,
                    confirmPassword
                  )
                ) {
                  pageChange("page-two");
                }
              }}
            >
              NEXT
            </Button>
          </div>
          <p className="flex justify-end mt-2 primary-blue-text">
            Already have an account?&nbsp;
            <Button
              anchorTag
              onClick={() => {
                setFirstName("");
                setLastName("");
                setEmail("");
                setPassword("");
                setConfirmPassword("");
                 navigate(-1);
              }}
            >
              Click Here
            </Button>
          </p>
        </div>
      </Container>
    </Page>
  );
}

export default PageOne
