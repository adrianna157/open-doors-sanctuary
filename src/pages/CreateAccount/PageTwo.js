import React, { useContext } from "react"
import { validatePageTwo } from '../../helpers/createAccount/createAccountValidation.js'
import HeaderText from '../../components/HeaderText'
import Button from '../../components/Button'
import TextField from '../../components/TextField'
import Page from '../../components/Page'
import Container from '../../components/Container'
import AuthHeader from '../../components/AuthHeader'
import DropDown from '../../components/DropDown'
import { NotificationContext } from "../../helpers/AlertContext/AlertContext.js";
import AlertHandler from "../../components/AlertHandler/index.js"

const PageTwo = (props) => {
  const [alert, setAlert] = useContext(NotificationContext);

  let {
    birthYear, setBirthYear,
    location, setLocation,
    agreeToTerms, setAgreeToTerms,
    pageChange, signUp,
    setFirstName, setLastName,
    setEmail, setPassword,
    setConfirmPassword,
    country, setCountry
  } = props

  const stateDropDownConditionalRender = () => {
    if (country.value === "US")  {
      return (
        <DropDown
          USAStates
          value={location}
          onChange={(value) => setLocation(value)}
          label="State"
          className="mb-4"
        />
      )
    }
  }



  return (
    <Page noLeftMargin>
      <AlertHandler />
      <AuthHeader />
      <Container
        grayedBackground
        height="md:h-[32rem] sm:h-1/6 lg:h-63pr"
        width="md:w-[32rem] sm:w-1/6 lg:w-1/3"
        padding="pt-3pr px-4pr sm:pl-2"
        className="flex flex-col"
        margin=" ml-16"
        mariginTop="mt-24 lg:mt-h26pr"
      >
        <center>
          <HeaderText noBold fontSize="text-4xl" className="pb-2pr">
            Create an Account
          </HeaderText>
        </center>
        <div>
          <TextField
            value={birthYear}
            onChange={(e) => setBirthYear(e.target.value)}
            placeHolder="YYYY"
            label="Birth Year"
            className="mb-4"
          />
          <DropDown
            Country
            value={country}
            onChange={(value) => setCountry(value)}
            label="Country"
            className="mb-4"
          />
          {stateDropDownConditionalRender()}
          <div className="flex justify-start my-3pr">
            <TextField
              noRounded
              noFullFieldWidth
              fieldWidth="w-5"
              fieldHeight="h-5"
              width="w-5"
              type="checkbox"
              checked={agreeToTerms}
              onChange={(e) => {
                setAgreeToTerms(e.target.checked);
              }}
            />
            <p className="mx-2 primary-blue-text">
              I agree to the&nbsp;
              <Button
                anchorTag
                onClick={() => {
                  pageChange("terms-of-use");
                }}
              >
                Terms of Use
              </Button>
            </p>
          </div>
          <div className="flex justify-between mt-h4pr">
            <Button
              linedBlue
              noPadding
              className="w-20pr lg:w-6pr"
              onClick={() => {
                setBirthYear("");
                setLocation("");
                setAgreeToTerms("");
                setFirstName("");
                setLastName("");
                setEmail("");
                setPassword("");
                setConfirmPassword("");
                pageChange("page-one");
                props.onStateChange("signIn", {});
              }}
            >
              CANCEL
            </Button>
            <Button
              solidBlue
              noPadding
              className="w-20pr lg:w-6pr"
              onClick={() => {
                if (
                  validatePageTwo(
                    setAlert,
                    birthYear,
                    location.abbreviation,
                    agreeToTerms,
                    country.value,
                    country.label
                  )
                ) {
                  signUp();
                }
              }}
            >
              SUBMIT
            </Button>
          </div>
        </div>
      </Container>
    </Page>
  );
}

export default PageTwo
