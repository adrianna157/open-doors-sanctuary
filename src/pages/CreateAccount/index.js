import React, { useState, Fragment, useContext } from "react"
import { Auth } from "aws-amplify"
import PageOne from './PageOne.js'
import PageTwo from './PageTwo.js'
import TermsModal from "./TermsModal.js";
import { AuthContext } from '../../helpers/AuthContext/AuthContext.js';
import randomWords from 'random-words'
import { NotificationContext } from "../../helpers/AlertContext/AlertContext.js";

const CreateAccount = (props) => {
  const [firstName, setFirstName] = useState("")
  const [lastName, setLastName] = useState("")
  const [email, setEmail] = useState("")
  const [password, setPassword] = useState("")
  const [confirmPassword, setConfirmPassword] = useState("")
  const [alias, setAlias] = useState(randomWords())
  const [birthYear, setBirthYear] = useState("")
  const [location, setLocation] = useState("")
  const [country, setCountry] = useState("")
  // TODO: ask about the "How did you hear about us"
  // Trevor will supply source list, and there will be an other field that will allow users to input a source
  const [agreeToTerms, setAgreeToTerms] = useState(false)
  const [pageState, setPageState] = useState(props.initialPage ? props.initialPage : "page-one")
  const [state, dispatch] = useContext(AuthContext)
  const [alert, setAlert] = useContext(NotificationContext);

  const signUp = () => {
    Auth.signUp({
      username: email.toLowerCase(),
      password,
      attributes: {
        email: email.toLowerCase(),
        nickname: alias,
        given_name: firstName,
        family_name: lastName,
        birthdate: "??/??/" + birthYear,
        locale: location.abbreviation,
        'custom:status': 'pending'
     } })
      .then((data) => {
        setFirstName("")
        setLastName("")
        setConfirmPassword("")
        setBirthYear("")
        setLocation("")
        props.onStateChange("confirmSignUp", {
          email: email.toLowerCase(),
          setEmail: setEmail,
          alias: alias,
          setAlias: setAlias,
          password: password,
          setPassword: setPassword
        })
      })
      .catch((err) => {
        setAlert({
          type: "SET_NOTIFICATION",
          payload: {
            occurs: true,
            message: err.message,
            textColor: "redText",
            borderColor: "redBorder",
          },
        });
      })
  }

  if (props.authState === 'signUp') {
    switch (pageState) {
      case "page-one":
        return (
          <PageOne
            firstName={firstName} setFirstName={setFirstName}
            lastName={lastName} setLastName={setLastName}
            email={email} setEmail={setEmail}
            password={password} setPassword={setPassword}
            confirmPassword={confirmPassword} setConfirmPassword={setConfirmPassword}
            pageChange={setPageState} onStateChange={props.onStateChange}
          />
        )
      case "page-two":
        return (
          <PageTwo
            setFirstName={setFirstName} setLastName={setLastName}
            setEmail={setEmail} setPassword={setPassword}
            setConfirmPassword={setConfirmPassword}
            alias={alias} setAlias={setAlias}
            birthYear={birthYear} setBirthYear={setBirthYear}
            country={country} setCountry={setCountry}
            location={location} setLocation={setLocation}
            agreeToTerms={agreeToTerms} setAgreeToTerms={setAgreeToTerms}
            pageChange={setPageState} onStateChange={props.onStateChange}
            signUp={signUp}
          />
        )
      case "terms-of-use":
        return (
          <TermsModal
            pageChange={setPageState}
            onStateChange={props.onStateChange}
          ></TermsModal>
        );
      default:
        return (
          <PageOne
            firstName={firstName} setFirstName={setFirstName}
            lastName={lastName} setLastName={setLastName}
            email={email} setEmail={setEmail}
            password={password} setPassword={setPassword}
            confirmPassword={confirmPassword} setConfirmPassword={setConfirmPassword}
            pageChange={setPageState} onStateChange={props.onStateChange}
          />
        )
    }
  } else {
    return <Fragment></Fragment>
  }
}

export default CreateAccount
