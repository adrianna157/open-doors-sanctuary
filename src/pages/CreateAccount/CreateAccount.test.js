import React from 'react'
import Store from "../../helpers/AuthContext/AuthContext.js"
import AlertStore from "../../helpers/AlertContext/AlertContext.js";
import renderer from "react-test-renderer";
import CreateAccount from "./index.js";

describe('The CreateAccount Page Renders Correctly', () => {
    it('Should Render CreateAccount Page Component Page 1', () => {
        const component = renderer.create(
          <Store>
            <AlertStore>
              <CreateAccount authState="signUp" />
            </AlertStore>
          </Store>
        ).toJSON();
        expect(component).toMatchSnapshot()
    });

    it('Should Render CreateAccount Page Component Page 2', () => {
        const component = renderer.create(
          <Store>
            <AlertStore>
              <CreateAccount authState="signUp" initialPage="page-two" />
            </AlertStore>
          </Store>
        );

        expect(component.toJSON()).toMatchSnapshot()
    });

    it("Should not Render CreateAccount Page Component", () => {
      const component = renderer
        .create(
          <Store>
            <AlertStore>
              <CreateAccount />
            </AlertStore>
          </Store>
        )
        .toJSON();
      expect(component).toMatchSnapshot();
    });
});