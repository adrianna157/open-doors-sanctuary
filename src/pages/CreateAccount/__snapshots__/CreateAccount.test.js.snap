// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The CreateAccount Page Renders Correctly Should Render CreateAccount Page Component Page 1 1`] = `
<div
  className="h-screen lg:h-auto lg:overflow-hidden  pb-6 mb-6"
>
  <div>
    <div
      className=" inset-y-h12pr -z-10 fixed  content-center flex-wrap h-46pr bg-no-repeat bg-cover bg-scroll w-screen bg-none  lg:flex lg:justify-center lg:bg-coin "
    />
    <div
      className="lg:flex lg:justify-center w-full"
    >
      <nav
        className="h-6pr bg-white fixed top-0 left-0 w-screen justify-center flex shadow-lg lg:h-12pr lg:justify-between lg:shadow-none"
      >
        <section
          className="h-4pr my-h08pr lg:flex lg:h-10pr lg:p-1"
        >
          <a
            href="/"
          >
            <img
              alt="Open Doors Sanctuary"
              className="h-4pr lg:h-10pr lg:ml-w05pr"
              src="guardian_group_logo.svg"
            />
          </a>
        </section>
      </nav>
      <div
        className="bg-white shadow-container lg:shadow-xl  w-96pr lg:w-35pr h-91pr lg:h-63pr mt-h8pr lg:mt-h26pr mx-auto rounded z-40 relative"
      >
        <div
          className="h-full standard-text-font container-shadow  pt-3pr px-4pr flex flex-col"
        >
          <center>
            <div
              className="mb-2 primary-blue-text font-display text-4xl pb-2pr"
            >
              Create an Account
            </div>
          </center>
          <div>
            <div
              className="w-full mb-4"
            >
              <div
                className="flex"
              >
                <p
                  className="light-primary-blue-text font-light mb-1"
                >
                  First Name
                </p>

              </div>
              <input
                className="textfield-bg-color py-h3pr lg:py-h1pr w-full rounded h-4pr pl-4"
                onChange={[Function]}
                placeholder="First Name"
                value=""
              />
            </div>
            <div
              className="w-full mb-4"
            >
              <div
                className="flex"
              >
                <p
                  className="light-primary-blue-text font-light mb-1"
                >
                  Last Name
                </p>

              </div>
              <input
                className="textfield-bg-color py-h3pr lg:py-h1pr w-full rounded h-4pr pl-4"
                onChange={[Function]}
                placeholder="Last Name"
                value=""
              />
            </div>
            <div
              className="w-full mb-4"
            >
              <div
                className="flex"
              >
                <p
                  className="light-primary-blue-text font-light mb-1"
                >
                  Email
                </p>

              </div>
              <input
                className="textfield-bg-color py-h3pr lg:py-h1pr w-full rounded h-4pr pl-4"
                onChange={[Function]}
                placeholder="Email"
                type="email"
                value=""
              />
            </div>
            <div
              className="w-full mb-4"
            >
              <div
                className="flex"
              >
                <p
                  className="light-primary-blue-text font-light mb-1"
                >
                  Password
                </p>
                <div
                  className="relative flex items-center"
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                >
                  <div
                    className="absolute whitespace-no-wrap primary-solid-blue to-gray-700 text-white px-4 py-2 rounded flex items-center transition-all duration-150"
                    style={
                      Object {
                        "left": "10%",
                        "opacity": 0,
                      }
                    }
                  >
                    <div
                      className="primary-solid-blue h-3 w-3 absolute"
                      style={
                        Object {
                          "left": "-6px",
                          "transform": "rotate(45deg)",
                        }
                      }
                    />
                    Your password must be 12 characters long, have at least one number, one upper case, one lower case, and one special character
                  </div>
                  <svg>
                    information.svg
                  </svg>
                </div>
              </div>
              <input
                className="textfield-bg-color py-h3pr lg:py-h1pr w-full rounded h-4pr pl-4"
                onChange={[Function]}
                placeholder="Password"
                type="password"
                value=""
              />
            </div>
            <div
              className="w-full mb-4pr"
            >
              <div
                className="flex"
              >
                <p
                  className="light-primary-blue-text font-light mb-1"
                >
                  Confirm Password
                </p>

              </div>
              <input
                className="textfield-bg-color py-h3pr lg:py-h1pr w-full rounded h-4pr pl-4"
                onChange={[Function]}
                placeholder="Password"
                type="password"
                value=""
              />
            </div>
            <div
              className="flex justify-between mb-h2pr"
            >
              <button
                className="btton primary-lined-blue w-20pr lg:w-6pr"
                onClick={[Function]}
              >
                CANCEL
              </button>
              <button
                className="btton primary-solid-blue w-20pr lg:w-6pr"
                onClick={[Function]}
              >
                NEXT
              </button>
            </div>
            <p
              className="flex justify-end mt-2 primary-blue-text"
            >
              Already have an account?Â 
              <button
                className="underline primary-blue-text"
                onClick={[Function]}
              >
                Click Here
              </button>
            </p>
          </div>
        </div>
      </div>
      <div
        className="container-grayed "
      />
    </div>
  </div>
</div>
`;

exports[`The CreateAccount Page Renders Correctly Should Render CreateAccount Page Component Page 2 1`] = `
<div
  className="h-screen lg:h-auto lg:overflow-hidden  pb-6 mb-6"
>
  <div>
    <div
      className=" inset-y-h12pr -z-10 fixed  content-center flex-wrap h-46pr bg-no-repeat bg-cover bg-scroll w-screen bg-none  lg:flex lg:justify-center lg:bg-coin "
    />
    <div
      className="lg:flex lg:justify-center w-full"
    >
      <nav
        className="h-6pr bg-white fixed top-0 left-0 w-screen justify-center flex shadow-lg lg:h-12pr lg:justify-between lg:shadow-none"
      >
        <section
          className="h-4pr my-h08pr lg:flex lg:h-10pr lg:p-1"
        >
          <a
            href="/"
          >
            <img
              alt="Open Doors Sanctuary"
              className="h-4pr lg:h-10pr lg:ml-w05pr"
              src="guardian_group_logo.svg"
            />
          </a>
        </section>
      </nav>
      <div
        className="bg-white shadow-container lg:shadow-xl  w-96pr lg:w-35pr h-91pr lg:h-63pr mt-h8pr lg:mt-h26pr mx-auto rounded z-40 relative"
      >
        <div
          className="h-full standard-text-font container-shadow  pt-3pr px-4pr flex flex-col"
        >
          <center>
            <div
              className="mb-2 primary-blue-text font-display text-4xl pb-2pr"
            >
              Create an Account
            </div>
          </center>
          <div>
            <div
              className="w-full mb-4"
            >
              <div
                className="flex"
              >
                <p
                  className="light-primary-blue-text font-light mb-1"
                >
                  Birth Year
                </p>

              </div>
              <input
                className="textfield-bg-color py-h3pr lg:py-h1pr w-full rounded h-4pr pl-4"
                onChange={[Function]}
                placeholder="YYYY"
                value=""
              />
            </div>
            <div
              className="w-full mb-4"
            >
              <p
                className="light-primary-blue-text font-light mb-1"
              >
                Country
              </p>
              <div
                className="w-full rounded h-4pr css-2b097c-container"
                onKeyDown={[Function]}
              >
                <span
                  aria-atomic="false"
                  aria-live="polite"
                  aria-relevant="additions text"
                  className="css-1f43avz-a11yText-A11yText"
                />
                <div
                  className=" css-yk16xz-control"
                  onMouseDown={[Function]}
                  onTouchEnd={[Function]}
                >
                  <div
                    className=" css-g1d714-ValueContainer"
                  >
                    <div
                      className=" css-1wa3eu0-placeholder"
                    >
                      Select...
                    </div>
                    <div
                      className="css-b8ldur-Input"
                    >
                      <div
                        className=""
                        style={
                          Object {
                            "display": "inline-block",
                          }
                        }
                      >
                        <input
                          aria-autocomplete="list"
                          autoCapitalize="none"
                          autoComplete="off"
                          autoCorrect="off"
                          disabled={false}
                          id="react-select-2-input"
                          onBlur={[Function]}
                          onChange={[Function]}
                          onFocus={[Function]}
                          spellCheck="false"
                          style={
                            Object {
                              "background": 0,
                              "border": 0,
                              "boxSizing": "content-box",
                              "color": "inherit",
                              "fontSize": "inherit",
                              "label": "input",
                              "opacity": 1,
                              "outline": 0,
                              "padding": 0,
                              "width": "1px",
                            }
                          }
                          tabIndex="0"
                          type="text"
                          value=""
                        />
                        <div
                          style={
                            Object {
                              "height": 0,
                              "left": 0,
                              "overflow": "scroll",
                              "position": "absolute",
                              "top": 0,
                              "visibility": "hidden",
                              "whiteSpace": "pre",
                            }
                          }
                        >

                        </div>
                      </div>
                    </div>
                  </div>
                  <div
                    className=" css-1hb7zxy-IndicatorsContainer"
                  >
                    <span
                      className=" css-1okebmr-indicatorSeparator"
                    />
                    <div
                      aria-hidden="true"
                      className=" css-tlfecz-indicatorContainer"
                      onMouseDown={[Function]}
                      onTouchEnd={[Function]}
                    >
                      <svg
                        aria-hidden="true"
                        className="css-tj5bde-Svg"
                        focusable="false"
                        height={20}
                        viewBox="0 0 20 20"
                        width={20}
                      >
                        <path
                          d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                        />
                      </svg>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              className="flex justify-start my-3pr"
            >
              <div
                className="w-5"
              >
                <div
                  className="flex"
                >


                </div>
                <input
                  checked={false}
                  className="textfield-bg-color py-h3pr lg:py-h1pr h-5 w-5 pl-4"
                  onChange={[Function]}
                  type="checkbox"
                />
              </div>
              <p
                className="mx-2 primary-blue-text"
              >
                I agree to theÂ 
                <button
                  className="underline primary-blue-text"
                  onClick={[Function]}
                >
                  Terms of Use
                </button>
              </p>
            </div>
            <div
              className="flex justify-between mt-h4pr"
            >
              <button
                className="btton primary-lined-blue w-20pr lg:w-6pr"
                onClick={[Function]}
              >
                CANCEL
              </button>
              <button
                className="btton primary-solid-blue w-20pr lg:w-6pr"
                onClick={[Function]}
              >
                SUBMIT
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        className="container-grayed "
      />
    </div>
  </div>
</div>
`;

exports[`The CreateAccount Page Renders Correctly Should not Render CreateAccount Page Component 1`] = `null`;
